invenio_rdm_records/services/communities/service.py

```diff
@@ -166,6 +168,10 @@ def _remove(self, identity, community_id, record):
         self.require_permission(
             identity, "remove_community", record=record, community_id=community_id
         )
+        if current_app.config.get("RDM_ENSURE_RECORD_COMMUNITY_EXISTS"):
```

Comment:
we have discussed this PR internally with my colleagues and I have some more comments to follow up (sorry about that!)

- this check will not ensure the record being part of at least one community, however you can still create a record via deposit form and select no community. Is this a desired behaviour for the feature? If yes, what is the use case for allowing to create a record without a community but not allowing to remove one, once the community is added at least once? Could you explain us a bit more in what cases is this desired behaviour?

- this part seems to have reflection only in the backend, it should also be reflected in the frontend if we would like to consider it as a full feature, meaning, for example if `RDM_ENSURE_RECORD_COMMUNITY_EXISTS` and there is only one community attached to a record, the `Remove` button should be blocked in the UI, with a toolltip informing about the reason of blocking the button. Otherwise, the user is not informed early about the limitations in the instance (only when the error appears after an attempt of removing the last community)

Reply:
Thank you for the valuable input @kpsherva.
- to document the answer to the first question, We are currently depending on [invenio-config-kth](https://pypi.org/project/invenio-config-kth/) that overrides permissions to prevent record publishing in the absence of a selected community.

After our discussion on Discord, we agreed to include this behavior in the current PR to make it a complete feature so that when this option is True, you as a user will be denied publishing a record without belonging to at least one community and denied the removal of the last record community.

- We have also decided to bring up this question during the next teleconference with the community to seek their feedback on what behavior we expect in edge cases like: 
> What happens if you have a restricted community attached and a public one and you try to remove the public one what should happen then?

> Only the community manager of the community in question can remove the record from a community. A manager of community A cannot remove the record from community B.
so if community B is restricted (and the community manager A is not part of community B = he cannot see it), but in the interface, the button to remove community A is not disabled, it means (from the point of the manager A) there is a hidden last community.

This can be related to the question if we should make the "remove community" button visible in the UI or not.

Reply:
Thank you for your suggestions. I see your perspectives on both approaches. However, before finalizing our decision, I would also like to hear what the CERN team suggests on this matter. Their input could potentially provide additional insights or even more suitable solutions.